File: students/students/src/MON.SPA/src/models/studentCreateModel.js
export class StudentModel {
  constructor(obj = {}) {
    this.id = obj.id || null;
    this.pin = obj.pin || '';
    this.publicEduNumber = obj.publicEduNumber || '';
    this.pinType = obj.pinType || null;
    this.firstName = obj.firstName || '';
    this.middleName = obj.middleName || '';
    this.lastName = obj.lastName || '';
    this.birthDate = obj.birthDate || '';
    this.gender = obj.gender && obj.gender.value || '';
    this.nationalityId = obj.nationalityId;
    this.birthPlaceCountryId = obj.birthPlaceCountryId;
    this.birthPlaceId = obj.birthPlaceId;
    this.permanentResidenceId = obj.permanentResidenceId;
    this.usualResidenceId = obj.usualResidenceId;
    this.currentAddress = obj.currentAddress || '';
    this.permanentAddress = obj.permanentAddress || '';
    this.phoneNumber = obj.phoneNumber || '';
    this.email = obj.email || '';
    this.addressCoincidesCheck = obj.addressCoincidesCheck || false;
    this.internationalProtectionStatus = obj.internationalProtectionStatus || false;
    this.hasIndividualStudyPlan = obj.hasIndividualStudyPlan || false;
    this.hasSupportiveEnvironment = obj.hasSupportiveEnvironment || false;
    this.supportiveEnvironment = obj.supportiveEnvironment || '';
    this.repeater = obj.repeater || '';
    this.birthPlace = obj.birthPlace || null;
    this.genderId = obj.genderId;
  }
}

File: students/students/src/MON.Models/StudentModels/StudentCreateModel.cs
namespace MON.Models.StudentModels
{
    using System;
    using System.ComponentModel.DataAnnotations;

    public class StudentCreateModel
    {
        /// <summary>
        /// Използва се при редакция
        /// </summary>
        public int? Id { get; set; }
        [Required]
        public string Pin { get; set; }
        [Required]
        public int PinTypeId { get; set; }
        [Required]
        public string FirstName { get; set; }
        public string MiddleName { get; set; }
        [Required]
        public string LastName { get; set; }
        public DateTime? BirthDate { get; set; }
        [Required]
        public int GenderId { get; set; }
        public int? BirthPlaceId { get; set; }
        public int? UsualResidenceId { get; set; }
        public int? PermanentResidenceId { get; set; }
        [Required]
        public string CurrentAddress { get; set; }
        [Required]
        public string PermanentAddress { get; set; }
        public string PhoneNumber { get; set; }
        [Required]
        public int? NationalityId { get; set; }
        [Required]
        public int? BirthPlaceCountryId { get; set; }
        public string BirthPlace { get; set; }
        public string Email { get; set; }
    }
}
